src/routeGeneration/templateHelpers.ts

```diff
@@ -359,6 +363,95 @@ export class ValidationService {
     return new Buffer(value);
   }
 
+  public validateUnion(name: string, value: any, fieldErrors: FieldErrors, subSchemas: TsoaRoute.PropertySchema[] | undefined, parent = ''): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors = subSchemas.map(subSchema => {
+      const subFieldErrors: FieldErrors = {};
+      this.ValidateParam(subSchema, value, name, subFieldErrors, parent);
+      return subFieldErrors;
+    });
+
+    if (!subFieldErrors.some(subFieldError => Object.keys(subFieldError).length === 0)) {
+      fieldErrors[parent + name] = {
+        message: `Could not match the union against any of the items. Issues: ${subFieldErrors}`,
+        value,
+      };
+      return;
+    }
+
+    return value;
+  }
+
+  public validateIntersection(name: string, value: any, fieldErrors: FieldErrors, subSchemas: TsoaRoute.PropertySchema[] | undefined, parent = ''): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors = subSchemas.map(subSchema => {
+      const subFieldErrors: FieldErrors = {};
+      this.ValidateParam(subSchema, value, name, subFieldErrors, parent);
+      return subFieldErrors;
+    });
+
+    subFieldErrors.forEach(subFieldError => {
+      (Object as any).entries(subFieldError).forEach(([key, value]) => {
+        if (value.message.includes('excess')) {
+          delete subFieldError[key];
+        }
+      });
+    });
+
+    if (subFieldErrors.some(subFieldError => Object.keys(subFieldError).length !== 0)) {
+      fieldErrors[parent + name] = {
+        message: `Could not match the intersection against every type. Issues: ${subFieldErrors}`,
+        value,
+      };
+      return;
+    }
+
+    // Only Model definitions make sense here right now
+    const refNames = subSchemas.filter(subschema => subschema.ref).map(subschema => subschema.ref) as string[];
+    if (!refNames.every(refName => this.models[refName])) {
+      return value;
+    }
+
+    const reportedExcess = refNames.map(refName => this.models[refName]).reduce((acc, subSchema) => {
+      return new Set([...acc, ...this.getExcessPropertiesFor(subSchema, new Set(Object.keys(value)))]);
+    }, new Set());
+
+    const allowedProperties = refNames.map(refName => this.models[refName]).reduce((acc, subSchema) => {
+      return new Set([...acc, ...this.getPropertiesFor(subSchema)]);
+    }, new Set());
+
+    const actualExcess = new Set([...reportedExcess].filter(property => !allowedProperties.has(property)));
```

Comment:
I don't see any of this code using `noImplicitAdditionalProperties` (I'm talking about the boolean version). If `noImplicitAdditionalProperties` is set to `false | undefined` then you don't even need to do the excess property check.

Reply:
In that case, `reportedExcess` will be an empty Set.